homeassistant/components/climate/flexit.py

```diff
@@ -0,0 +1,274 @@
+        self._heater_enabled = \
+            bool(self._input_regs['HeatingBatteryActive']['value'])
+        # Current operation mode
+        if self._heating:
+            self._current_operation = 'Heating'
+        elif self._cooling:
+            self._current_operation = 'Cooling'
+        elif self._heat_recovery:
+            self._current_operation = 'Recovering'
+        elif self._input_regs['ActualSetAirSpeed']['value']:
+            self._current_operation = 'Fan Only'
+        else:
+            self._current_operation = 'Off'
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            'filter_hours':     self._filter_hours,
+            'filter_alarm':     self._filter_alarm,
+            'heat_recovery':    self._heat_recovery,
+            'heating':          self._heating,
+            'heater_enabled':   self._heater_enabled,
+            'cooling':          self._cooling
+        }
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._current_operation
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        return self._current_fan_mode
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return self._fan_list
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
+
+        modbus.HUB.write_register(
+            unit=self._slave,
+            address=(self._holding_regs['SetAirTemperature']['addr']),
+            value=round(self._target_temperature * 10.0))
+        _LOGGER.debug("Setting temperature for %s to %f",
+                      self._name, self._target_temperature)
+        self.schedule_update_ha_state()
+
+    def set_fan_mode(self, fan):
+        """Set new fan mode."""
+        modbus.HUB.write_register(
+            unit=self._slave,
+            address=(self._holding_regs['SetAirSpeed']['addr']),
+            value=self._fan_list.index(fan))
+        self._current_fan_mode = self._fan_list.index(fan)
+        self.schedule_update_ha_state()
```

Comment:
No need to call these as it's done automatically because should_poll is returning True.